community/eigen/APKBUILD

```diff
@@ -1,36 +1,46 @@
+# Contributor: stef <l0ls0fo2i@ctrlc.hu>
 # Contributor: Bradley J Chambers <brad.chambers@gmail.com>
 # Maintainer: Bradley J Chambers <brad.chambers@gmail.com>
 pkgname=eigen
 pkgver=3.3.7
-pkgrel=0
+pkgrel=1
 pkgdesc="Eigen is a C++ template library for linear algebra"
 url="http://eigen.tuxfamily.org/index.php?title=Main_Page"
 arch="noarch"
-options="!check"  # Headers-only: no tests are possible.
 license="MPL-2.0"
 depends=""
-makedepends=""
+options="!check"  # checks take a long time, and tend to fail randomly
+makedepends="cmake openblas-dev suitesparse-dev mpfr-dev fftw-dev boost-dev gmp-dev glu-dev freeglut-dev"
 install=""
 subpackages="$pkgname-dev $pkgname-doc"
-source="$pkgname-$pkgver.tar.gz::http://bitbucket.org/eigen/$pkgname/get/$pkgver.tar.gz"
+source="$pkgname-$pkgver.tar.gz::http://bitbucket.org/eigen/$pkgname/get/$pkgver.tar.gz
+	fftw-test-fix.patch"
 builddir="$srcdir/$pkgname-$pkgname"
 
 build() {
-        return 0
+	mkdir -p build
+	cd build
+	cmake -DCMAKE_INSTALL_PREFIX=/usr "$builddir"
+	make
 }
 
 prepare() {
-        mv "$builddir"-* "$builddir"  # directory name contains hash
-        default_prepare
+	mv "$builddir"-* "$builddir"  # directory name contains hash
+	default_prepare
+}
+
+check() {
+	cd "$builddir/build"
+	make check
 }
 
 package() {
-        mkdir -p "$pkgdir"/usr/include/eigen3
-        cp -r "$builddir"/Eigen "$pkgdir"/usr/include/eigen3
-        cp -r "$builddir"/unsupported "$pkgdir"/usr/include/eigen3
-        cp "$builddir"/signature_of_eigen3_matrix_library "$pkgdir"/usr/include/eigen3
-        mkdir -p "$pkgdir"/usr/share/doc/$pkgname
-        cp -r "$builddir"/doc/ "$pkgdir"/usr/share/doc/$pkgname
+	cd "$builddir/build"
+	make DESTDIR="$pkgdir" install
+	mkdir -p "$pkgdir"/usr/share/doc/$pkgname
+	cp -r "$builddir"/doc/ "$pkgdir"/usr/share/doc/$pkgname
+	install -Dm0644 eigen3.pc "$pkgdir/usr/lib/pkgconfig/eigen3.pc"
```

Comment:
Huh, CMake doesn't automatically install this?

Reply:
uh, yeah, it does. however it gets bundled in the main pkg, not in -dev. i changed it so it gets into -dev. however the main package is now empty. but i dunno how to disable the creation of an empty apk file. would be nice to suppress it.

Reply:
It's fine as-is.

Reply:
really? i would expect cmake and pkgconf things go into -dev, makes little sense to have them in the main pkg.

Reply:
Ah sorry if I wasn't clear, I mean it's fine `as-is` == `the way you're doing it right now`. It'll be clearer what you're trying to do once you've moved things into the `dev()` function :)

Reply:
i've moved it to a 'dev()' function and force pushed an update.